### PR TITLE
Bug/serializer stack overflow

### PR DESCRIPTION
**Pull request by jjelschen**

### CODE REVIEW

File: org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend [hunk 1]
```diff
@@ -66,18 +35,59 @@ abstract class AbstractXtextIntegrationTest {
...
+		ensureServiceLocatorPopulated
+		dropwizardServerRule = new EagerDropwizardClientRule(dummyResource)
+
+		return dropwizardServerRule
+	}
+
+	private def createClientRule() {
+		dropwizardClientRule = new DropwizardAppRule(TestEditorApplication,
+			ResourceHelpers.resourceFilePath('test-config.yml'),
+			#[config('indexServiceURL', '''«createServerRule.baseUri»/xtext/index/global-scope''')])
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend]: That the server rule is created here and is setting dropwizardServerRule is a bit hidden here.
How a method 'initializeRules' that creates the rules and initializes the variables.

File: org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend [hunk 2]
```diff
@@ -66,18 +35,59 @@ abstract class AbstractXtextIntegrationTest {
...
+	/**
+	 * Actually a client rule, but acts as the server (remote) for this test
+	 */
 	@Rule
-	public val dropwizardAppRule = new DropwizardAppRule(
-		TestEditorTestApplication,
-		ResourceHelpers.resourceFilePath('test-config.yml'),
-		configs
-	)
+	public var DropwizardClientRule dropwizardServerRule
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend]: how about naming this dropwizardIndexServerRule

File: org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend [hunk 3]
```diff
@@ -66,18 +35,59 @@ abstract class AbstractXtextIntegrationTest {
...
+
+	@Rule
+	public var DropwizardAppRule<TestEditorConfiguration> dropwizardClientRule = createClientRule
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend]: how about 'dropwizardXtextClientRule'?

File: org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend [hunk 4]
```diff
@@ -87,79 +97,13 @@ abstract class AbstractXtextIntegrationTest {
...
-			this.tslInjector = injectors.get(0)
-			this.tclInjector = injectors.get(1)
-			this.amlInjector = injectors.get(2)
-		}
-
-		override protected getLanguageSetups() {
-		}
+}
+
+class EagerDropwizardClientRule extends DropwizardClientRule {
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend]: a short explanation would be helpful (comment)

File: org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientRegressionTests.xtend [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	 * 
+	 * Observed behavior: in the context of the following request, the index service client tried to serialize the
+	 * Xtext resource's contents to send it as the body of the request to the index service. This seemed to throw off
+	 * the Xtext serializer.
+	 */
+	@Test
+	def void shouldNotCauseXtextSerializationToFail() {
+		// given
+		val tcl = '''
+			package org.testeditor.demo.swing
```
Comment (gunther-bachmann) [inline org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientRegressionTests.xtend:22]: tcl should be shortened (most of it is not really relevant for the test)

Comment (jjelschen) [reply org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientRegressionTests.xtend:22]: done (in a way that still reproduces the original issue as it did at 12709d2adc1423ef1f4a9bf44159aa96e66a5035)

**Pull request merged**